kotest-property/src/commonMain/kotlin/io/kotest/property/arbitraries/ints.kt

```diff
@@ -0,0 +1,97 @@
+package io.kotest.property.arbitraries
+
+import io.kotest.property.Arbitrary
+import io.kotest.property.PropertyInput
+import io.kotest.property.filter
+import io.kotest.property.Shrinker
+import io.kotest.property.withEdgeCases
+import kotlin.math.abs
+import kotlin.random.Random
+import kotlin.random.nextLong
+
+
+fun Arbitrary.Companion.int(
+   iterations: Int,
+   range: IntRange = Int.MIN_VALUE..Int.MAX_VALUE,
+   distribution: IntDistribution = IntDistribution.Uniform
+) = object : Arbitrary<Int> {
+   override fun edgecases(): Iterable<Int> = listOf(Int.MIN_VALUE, Int.MAX_VALUE, 0)
+   override fun samples(random: Random): Sequence<PropertyInput<Int>> {
+      return sequence {
+         for (k in 0 until iterations) {
+            val block = distribution.get(k, iterations, range.first.toLong()..range.last.toLong())
+            val next = random.nextLong(block).toInt()
+            val input = PropertyInput(next, IntShrinker)
+            yield(input)
+         }
+      }
+   }
+}
+
+/**
+ * Returns an [Arbitrary] where each value is a randomly chosen positive integer.
+ * The edge cases are: [Int.MAX_VALUE]
+ */
+fun Arbitrary.Companion.positiveIntegers(iterations: Int): Arbitrary<Int> =
+   int(iterations).withEdgeCases(Int.MAX_VALUE).filter { it > 0 }
```

Comment:
Using `filter` here might take too long to generate compatible values, and tests that use this will take longer to run

Reply:
Same impl that we had before.

Reply:
Plus it'll only through away 50% of values. If it was trying to find 1% of values it would be horrible.